src/__tests__/evaluator.test.ts

```diff
@@ -77,4 +85,157 @@ describe('Evaluator', () => {
 
     expect(actual_variation).toEqual('wanted_value');
   });
+
+  test.each([
+    // if (target.attr.email endswith '@harness.io' && target.attr.role = 'developer')
+    ['email_is_dev', 'boolflag_and', 'user@harness.io', 'developer', true],
+    ['email_is_mgr', 'boolflag_and', 'user@harness.io', 'manager', false],
+    ['ext_email_is_dev', 'boolflag_and', 'user@gmail.com', 'developer', false],
+    ['ext_email_is_mgr', 'boolflag_and', 'user@gmail.com', 'manager', false],
+    // if (target.attr.email endswith '@harness.io' || target.attr.email endswith '@somethingelse.com')
+    ['email_is_harness', 'boolflag_or', 'user@harness.io', 'n/a', true],
+    ['email_is_other', 'boolflag_or', 'user@somethingelse.com', 'n/a', true],
+    ['email_is_gmail', 'boolflag_or', 'user@gmail.com', 'n/a', false],
+  ])('Target rules v2 %s_%s', async (name, flagName, email, role, expected) => {
```

Comment:
Try to give the test human readable names so it's easier to track down without having to dig through the test code. If you want to have a section called `Target rules v2`, maybe consider adding a `describe` block around this.
```suggestion
  ])('it should do X when Y', async (name, flagName, email, role, expected) => {
```

You can play around with the ordering of the variables to get the `%s`s to line up properly, or add a short description of what you're trying to prove as the first element in the arrays and use that in the test name.